### PR TITLE
Add password resets

### PR DESCRIPTION
**Pull request by acron0**

### CODE REVIEW

File: src/witan/gateway/handler.clj [hunk 1]
```diff
@@ -64,10 +71,7 @@
   ([components path]
    (post-to-heimdall components path nil))
   ([components path params']
-   (let [params (if (and params' (or (= (type params') ByteArrayInputStream)
-                                     (= (type params') org.httpkit.BytesInputStream)))
-                  (transit-decode-bytes params')
-                  params')
+   (let [params (decode-params params')
```
Comment (otfrom) [inline src/witan/gateway/handler.clj]: I'd generally have `foo` and `foo'` the other way around.

File: src/witan/gateway/handler.clj [hunk 2]
```diff
@@ -40,6 +40,13 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;; Helper functions
 
+(defn decode-params
+  [params']
```
Comment (otfrom) [inline src/witan/gateway/handler.clj]: `params'` -> `params`

File: src/witan/gateway/handler.clj [hunk 3]
```diff
@@ -227,10 +231,29 @@
...
+  [req]
+  (let [comms    (get-in req [:components :comms])
+        username (:username (decode-params (:body req)))]
+    (if (clojure.string/blank? username)
+      {:status 400 :body (transit-encode "No username?")}
+      (do
+        (comms/send-command! comms
+                             :kixi.heimdall/create-password-reset-request
+                             "1.0.0"
+                             nil
```
Comment (otfrom) [inline src/witan/gateway/handler.clj]: passing in `nil` always makes me twitchy.

Comment (acron0) [inline src/witan/gateway/handler.clj]: So, this is because there's no user associated with this command

**Pull request merged**